File: solutions/dart/secret-handshake/1/lib/secret_handshake.dart
class SecretHandshake {
  List<String> commands(int number) {

    List<String> results = [];

    if (number & 1 == 1) results.add('wink');
    if (number & 2 == 2) results.add('double blink');
    if (number & 4 == 4) results.add('close your eyes');
    if (number & 8 == 8) results.add('jump');

    if (number & 16 == 16) results = results.reversed.toList();

    return results;
  }
}

File: solutions/python/secret-handshake/1/secret_handshake.py
def commands(binary_str):
    results=[]

    if binary_str[4] =='1':
        results.append('wink')
    if binary_str[3] =='1':
        results.append('double blink')
    if binary_str[2] =='1':
        results.append('close your eyes')
    if binary_str[1] =='1': 
        results.append('jump')

    if binary_str[0] == '1':
        results = list(reversed(results))

    return results


    


    